test-app.gradle

```diff
@@ -93,17 +123,55 @@ ext.applyTestAppModule = { Project project ->
     def generatedResDir = file("${buildDir}/generated/rncli/src/main/res/")
     generatedResDir.mkdirs()
 
+    // https://github.com/facebook/react-native/blob/3dfedbc1aec18a4255e126fde96d5dc7b1271ea7/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/common/assets/ReactFontManager.java#L25
+    def generatedFontsDir = file("${generatedAssetsDir}/fonts")
+    generatedFontsDir.mkdirs()
+
     def generatedRawDir = file("${generatedResDir}/raw")
     generatedRawDir.mkdirs()
 
     preBuild.dependsOn(tasks.register("copyAssets", Copy) {
         androidResourceFiles.each {
-            from(it)
+            if (!isFontFile(it) && !isMediaFile(it)) {
+                from(it)
+            }
         }
 
         into(generatedAssetsDir)
     })
 
+    preBuild.dependsOn(tasks.register("copyFonts", Copy) {
+        androidResourceFiles.each {
+            if (isFontFile(it)) {
+                from(it) {
+                    // File-based resource names must contain only lowercase
+                    // a-z, 0-9, or underscore
+                    rename {
+                        it.toLowerCase()
```

Comment:
(following up from my testing comment - if this is something we can't work around, we should have in the docs somewhere explaining that they will need to keep it in mind when importing a font, and maybe provide a snippet for how to handle different platforms in the fonts styling)

Reply:
fixed by latest commit 👍